### ISSUE TITLE
UPBGE does not recognize a class defined in a sub-module

### ISSUE THREAD

**Issue by mysticfall** [UPBGE/upbge#1230]
When I create a Python module in the same directory with the `.blend file`, classes defined in the module can be referenced from the Logic Editor without a problem.

However, if I move the same file under a sub-module, or a package, it fails to recognize the classes as valid BGE components.

I have created a minimal reproducible project to demonstrate the issue:
[namespace-issue.tar.gz](https://github.com/UPBGE/upbge/files/4808784/namespace-issue.tar.gz)

You can add `module.MyComponent` to an object from the Logic Editor without a problem. But if you try to add `mymodule.submodule.MyComponent`, it fails with the following message:
```
A submodule type was found, but it was not a valid subclass of KX_PythonComponent.
```
`module.py` and `module/submodule.py` have the exact same content, and `module/__init__.py` contains the following line:
```python
from mymodule import submodule
```

**Comment by youle31** [comment]
It was working in 2.7...

**Comment by youle31** [comment]
To test the branch above, the component name has just to be:

submodule.MyComponent

**Comment by youle31** [comment]
Fixed here: https://github.com/UPBGE/upbge/commit/49802a5edc3c820fad7f7226e7be4d1a3012d628

**Comment by youle31** [comment]
arf, actually, when you use the button create with last upbge you can create something named a.b and it will work, then the roblem might not be solved... But you can just import your script in the text editor and then register the component...

**Comment by youle31** [comment]
Let's just say that python components logic were not designed to be used with python packages or namespaces.
You can use this kindof architecture with logic bricks (python controller), but this doesn't work with python components.

**Comment by marechal-p** [comment]
PRs welcome.

**Comment by mysticfall** [comment]
Sorry, but I feel a bit confused with the closing and reopening of this issue. The original problem was about using a submodule like `a.b.C`, not a module itself, e.g. `a.B`.

So, I don't understand the suggestion to test `submodule.MyComponent` with the latest branch, or the reason why the issue was closed by the change and reopened again.

Has anything regarding the submodules behaviour (or lack thereof) in the latest branch?

**Comment by youle31** [comment]
I thought i had solved the issue here: https://github.com/UPBGE/upbge/commit/49802a5edc3c820fad7f7226e7be4d1a3012d628

But actually, I was wrong. The issue is still there. It is in a part of the code written in C and too complex for me. (the file name is python_component.c)

@marechal-p reopened the issue...

We just need someone who'd like to work on it.

**Comment by mysticfall** [comment]
@youle31 Thanks for the clarification! I hope we could find someone to work on this issue then.

**Comment by youle31** [comment]
Fixed by @mysticfall  here: 8f89bfe13ba7b30f88e07d . Thanks!
